dot/parachain/types/bitvec.go

```diff
@@ -26,26 +29,12 @@ func NewBitVec(bits []bool) BitVec {
 	}
 }
 
-// Bits returns the bits in the BitVec
-func (bv *BitVec) Bits() []bool {
-	return bv.bits
-}
-
-// Bytes returns the byte representation of the BitVec.Bits
-func (bv *BitVec) Bytes() []byte {
-	return bitsToBytes(bv.bits)
-}
-
-// Size returns the number of bits in the BitVec
-func (bv *BitVec) Size() uint {
-	return uint(len(bv.bits))
-}
-
 // bitsToBytes converts a slice of bits to a slice of bytes
 // Uses lsb ordering
 // TODO: Implement msb ordering
 // https://github.com/ChainSafe/gossamer/issues/3248
-func bitsToBytes(bits []bool) []byte {
+func (bv BitVec) bitsToBytes() []byte {
```

Comment:
There is really no need for this change. I prefer the previous way better of passing bits, since bitToBytes is a rather general function. It also doesn't make changes to bv.

Reply:
I would prefer to use the new approach because it follows a convention that we are transforming a type `X` into bytes, like we do with the the `common.Hash` and its function `ToBytes`, it does not change the underling hash but transform it to raw bytes. This approach is wildly used in our codebase

Reply:
I made it a receiver function since it modifies the internal state of `bv`, and it was a private function in the previous code.

Reply:
Given `bitsToBytes` was a private function and only used by passing in `BitVec.bits` I thought it made sense to make it a receiver function of `BitVec`.

Reply:
> I would prefer to use the new approach because it follows a convention that we are transforming a type X into bytes, like we do with the the common.Hash and its function ToBytes, it does not change the underling hash but transform it to raw bytes. This approach is wildly used in our codebase

If you plan to have it only limited to this struct, it's better to rename this to `toBytes`. 
`bitsToBytes` is something that we would name a function.